design-proposals/status-for-nncp-not-matching-any-node.md

```diff
@@ -0,0 +1,192 @@
+# Status for NNCP not matching any node(s)
+
+## Summary
+
+This proposal is trying to figure out a way on how to handle the status response kubernetes-nmstate is reporting on
+`oc get nncp -oyaml` when the NNCP is not matching any node(s).
+
+## Motivation
+
+Users unfamiliar with the status reports of `oc get nncp -oyaml` might confuse the status report of `status.conditions[].type`
+(Available) and incorrectly assume that a NNCP has been properly applied to node(s) even though it is possibly not
+matching any node(s) `status.conditions[].message` (Policy does not match any node) and `status.conditions[].reason`
+(NoMatchingNode). Tickets like [these](https://issues.redhat.com/browse/OPNET-81) have already come up in the past and
+should therefore be addressed to provide a clear answer.
+
+### User Stories
+
+- As a cluster administrator, I want to be able to easily judge if an NNCP has successfully been rolled out by the status.
+
+### Goals
+
+- Clarify which `status.condition[]` list an applied NNCP not matching any node(s) should have.
+
+### Non-Goals
+
+- Changes to `status.condition[]` on anything else than applied NNCP.
+
+## Proposal
+
+### Solution 1: Add new status [completed/ignored]
```

Comment:
Let's add a Solution 2 where we just has a new Reason "NoMatchingNodes" 

So users can just do a `kubectl wait --for=condition=Available` and when it timeouts they will see that reason == NoMatchinNodes and will be clear what's the issue.

They can also wait first for progressing=False and then check avaiable=True <- if this is not the case they will see NoMatchingNodes.

btw I prefer this options since it means we don't add just another condition that complicate the conditions matrix.